src/Interpreters/LogicalExpressionsOptimizer.cpp

```diff
@@ -149,7 +149,7 @@ void LogicalExpressionsOptimizer::collectDisjunctiveEqualityChains()
         {
             if (from_node != nullptr)
             {
-                auto res = or_parent_map.insert(std::make_pair(function, ParentNodes{from_node}));
+                auto res = or_parent_map.insert({function, ParentNodes{from_node}});
```

Comment:
`emplace` is surely able to replace `insert` (and `push_back`) in many cases. I would need to check if there is some automatic transform that does that (e.g. clang-tidy). To not blow PR further up, I would prefer to defer that to a separate PR.